pkg/actuators/machineset/controller.go

```diff
@@ -81,6 +86,20 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	if !machineSet.DeletionTimestamp.IsZero() {
 		return ctrl.Result{}, nil
 	}
+
+	if r.Gate.Enabled(featuregate.Feature(openshiftfeatures.FeatureGateMachineAPIMigration)) {
+		// The paused condition is set by the machine-api-operator's machineset controller
+		// here: https://github.com/openshift/machine-api-operator/blob/ea46700bc132f1ba40d91f523f4b3a16dfafe8b8/pkg/controller/machineset/controller.go#L176-L231,
+		// once it observes the machineset .status.authoritativeAPI is not MachineAPI.
+		// This is a minor controller meant for annotations, so starts skipping further action
+		// only when the main controller sets the paused condition.
+		// Context for this decision here: https://redhat-internal.slack.com/archives/GE2HQ9QP4/p1747830982876869
+		if conditions.IsTrue(machineSet, machine.PausedCondition) {
+			klog.V(3).Infof("%v: machine set has paused condition, taking no further action", machineSet.Name)
+			return ctrl.Result{}, nil
+		}
+	}
```

Comment:
This is effectively policy. It may be relatively simple, but we should centralise it for when we inevitably need to update it.

Reply:
We would need to open PRs against all the provider anyway to bump the central module (machine-api-operator) to this. I don't think it'd buy us much.

Reply:
It would be 2 extra PRs in total

Reply:
Given the size of this, I don't think we need a shared util on this. It's already using several utils and only making two decisions based on the output of those utils